src/Analyzer.JsonRuleEngine/Operators/InOperator.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators
+{
+    /// <summary>
+    /// An operator that evaluates the "in" JSON expression.
+    /// </summary>
+    internal class InOperator : LeafExpressionOperator
+    {
+        /// <summary>
+        /// Gets the name of this operator.
+        /// </summary>
+        public override string Name => "In";
+
+        /// <summary>
+        /// Creates an InOperator.
+        /// </summary>
+        /// <param name="specifiedValue">The value specified in the JSON rule, it has to be an array of basic JSON values.</param>
+        /// <param name="isNegative">Whether the result of EvaluateExpression() should be negated or not.</param>
+        public InOperator(JToken specifiedValue, bool isNegative)
+        {
+            this.SpecifiedValue = specifiedValue;
```

Comment:
There should be validation somewhere that what's passed in is actually a JArray.  The earlier this validation can happen in the runtime, the better.  If it can be enforced as part of the schema, that would be best.

Reply:
I'm thinking on adding it in LeafExpressionDefinition, I couldn't find an earlier stage to make the validation, what do you think?